adsingestp/parsers/elsevier.py

```diff
@@ -0,0 +1,360 @@
+                month = utils.MONTH_TO_NUMBER[month_name]
+                self.base_metadata["pubdate_print"] = year + "-" + month + "-00"
+            elif regex_ddmmmyy.match(pubdate):
+                pubdate_parts = pubdate.split()
+                year = pubdate_parts[2]
+                month_name = pubdate_parts[1][0:3].lower()
+                month = utils.MONTH_TO_NUMBER[month_name]
+                day = pubdate_parts[0]
+                if len(day) == 1:
+                    day = "0" + day
+                self.base_metadata["pubdate_print"] = year + "-" + month + "-" + day
+            elif regex_yyyy.match(pubdate):
+                year = pubdate
+                self.base_metadata["pubdate_print"] = year + "-00-00"
+
+    def _parse_edhistory(self):
+        # key: xml tag, value: self.base_metadata key
+        dates_trans = {
+            "ce:date-received": "edhist_rec",
+            "ce:date-revised": "edhist_rev",
+            "ce:date-accepted": "edhist_acc",
+        }
+
+        for date_xml in dates_trans.keys():
+            if self.record_meta.find(date_xml):
+                dates = self.record_meta.find_all(date_xml)
+                dates_out = []
+                for date in dates:
+                    # don't assign without at least a year
+                    if date.get("year"):
+                        month = date.get("month", "00")
+                        day = date.get("day", "00")
+                        dates_out.append(
+                            date.get("year")
+                            + "-"
+                            + (month if len(month) == 2 else ("0" + month))
+                            + "-"
+                            + (day if len(day) == 2 else ("0" + day))
+                        )
+                if date_xml == "ce:date-accepted":
+                    # this only accepts a single date, the other two accept a list
+                    dates_out = dates_out[0]
+
+                self.base_metadata[dates_trans[date_xml]] = dates_out
+
+    def _parse_title_abstract(self):
+        if self.record_meta.find("ce:title"):
+            self.base_metadata["title"] = self.record_meta.find("ce:title").get_text()
+        elif self.record_header.find("dct:title"):
+            self.base_metadata["title"] = self.record_header.find("dct:title").get_text()
+        elif self.record_meta.find("cd:textfn"):
+            self.base_metadata["title"] = self.record_meta.find("cd:textfn").get_text()
+
+        if self.record_meta.find("ce:subtitle"):
+            self.base_metadata["subtitle"] = self.record_meta.find("ce:subtitle").get_text()
+
+        if self.record_meta.find("ce:abstract"):
+            abstract = ""
+            abs_all = self.record_meta.find_all("ce:abstract")
+            for abs in abs_all:
+                if abs.find("ce:section-title"):
+                    if abs.find("ce:section-title").get_text().lower() == "abstract":
```

Comment:
We can avoid repeating a `find` and then a `find_all`, and also the two `find` on the same name by storing the result before the `if`. The same could be done in other parts of the code (I am not writing a comment for each to avoid being repetitive).